### PR TITLE
fix: the system allows adding one more item than the maximum limit

### LINKED ISSUE #308: The system allows adding one more item than the maximum limit
## Description

When landlords add a new `ShareHouse`, `Category`, `Task`, or `Tenant` to a share house, it is possible to exceed the maximum limit by one. For example, even if the maximum limit for `ShareHouse` is 10, an 11th share house can still be added.

## To Reproduce

1. Keep adding new categories until the limit
2. Try to add 1 more even though it's exceeding the limit

## Screenshots or videos

[Tascurator-more-than-limit.webm](https://github.com/user-attachments/assets/8fe6011c-fddf-443d-ba01-66ae0da3f7c8)

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

Fix the condition for the maximum number of each item to prevent creating items that exceed the maximum limit.

## Changes

- Fix the condition for the maximum number of category
- Fix the condition for the maximum number of task
- Fix the condition for the maximum number of tenant

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [x] Code readability and simplicity
- [x] Follows best practices and coding standards
- [x] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/routes/tenant.route.ts [hunk 1]
```diff
@@ -229,6 +234,25 @@ const app = new Hono()
...
+            {
+              error: SERVER_ERROR_MESSAGES.MAX_LIMIT_REACHED(
+                'tenants',
+                CONSTRAINTS.TENANT_MAX_AMOUNT,
+              ),
+            },
+            400,
+          );
+        }
+        console.log('tenant', tenants);
```
Comment (akiomatic) [inline src/app/api/[[...route]]/routes/tenant.route.ts]: Could you remove this console.log please?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you check 1 comment?

File: src/app/api/[[...route]]/routes/tenant.route.ts (hunk 1 above)
Comment (KaiKoide) [reply src/app/api/[[...route]]/routes/tenant.route.ts]: I removed that.

**Review by akiomatic — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**